xarray/tests/test_dataarray.py

```diff
@@ -1836,100 +1836,267 @@ def test_resample(self):
-        array = DataArray(np.ones(10), [('time', times)])
-        array[1] = np.nan
+        # Nearest
+        rs = array.resample(time='3H')
+        actual = rs.nearest()
+        new_times = rs._full_index
+        expected = DataArray(
+            array.reindex(time=new_times, method='nearest')
+        )
+        self.assertDataArrayIdentical(expected, actual)
+
+    def test_upsample_nd(self):
+        # Same as before, but now we try on multi-dimensional DataArrays.
+        xs = np.arange(6)
+        ys = np.arange(3)
+        times = pd.date_range('2000-01-01', freq='6H', periods=5)
+        data = np.tile(np.arange(5), (6, 3, 1))
+        array = DataArray(data,
+                          {'time': times, 'x': xs, 'y': ys},
+                          ('x', 'y', 'time'))
+
+        # Forward-fill
+        actual = array.resample(time='3H').ffill()
+        expected_data = np.repeat(data, 2, axis=-1)
+        expected_times = times.to_series().resample('3H').asfreq().index
+        expected_data = expected_data[..., :len(expected_times)]
+        expected = DataArray(expected_data,
+                             {'time': expected_times, 'x': xs, 'y': ys},
+                             ('x', 'y', 'time'))
+        self.assertDataArrayIdentical(expected, actual)
+
+        # Backward-fill
+        actual = array.resample(time='3H').ffill()
+        expected_data = np.repeat(np.flip(data, axis=-1), 2, axis=-1)
+        expected_data = np.flip(expected_data, axis=-1)
+        expected_times = times.to_series().resample('3H').asfreq().index
+        expected_data = expected_data[..., :len(expected_times)]
+        expected = DataArray(expected_data,
+                             {'time': expected_times, 'x': xs, 'y': ys},
+                             ('x', 'y', 'time'))
+        self.assertDataArrayIdentical(expected, actual)
+
+        # As frequency
+        actual = array.resample(time='3H').asfreq()
+        expected_data = np.repeat(data, 2, axis=-1).astype(float)[..., :-1]
+        expected_data[..., 1::2] = np.nan
+        expected_times = times.to_series().resample('3H').asfreq().index
+        expected = DataArray(expected_data,
+                             {'time': expected_times, 'x': xs, 'y': ys},
+                             ('x', 'y', 'time'))
+        self.assertDataArrayIdentical(expected, actual)
+
+        # Pad
+        actual = array.resample(time='3H').pad()
+        expected_data = np.repeat(data, 2, axis=-1)
+        expected_data[..., 1::2] = expected_data[..., ::2]
+        expected_data = expected_data[..., :-1]
+        expected_times = times.to_series().resample('3H').asfreq().index
+        expected = DataArray(expected_data,
+                             {'time': expected_times, 'x': xs, 'y': ys},
+                             ('x', 'y', 'time'))
+        self.assertDataArrayIdentical(expected, actual)
+
+    def test_upsample_interpolate(self):
```

Comment:
this needs a `@requires_scipy` decorator or put

```Python
interp1d = pytest.importorskip('scipy.interpolate.interp1d')
```
in the first line of the test (instead of the import)